index.js

```diff
@@ -131,6 +133,10 @@ module.exports._getDependencies = function(config) {
 
     debug('cabinet result ' + result);
 
+    if (!result && !config.nonExistent[dep]) {
```

Comment:
Since `config.nonExistent` is an array and dep is a string (not an index), `config.nonExistent[dep]` will always be falsy.

I saw your comment on putting this with `!exists`, that sounds like a good approach. I don't see why `!exists` wouldn't be true for your test case's non-existent file.